libnetwork/networkdb/watch.go

```diff
@@ -41,25 +48,26 @@ func (nDB *NetworkDB) Watch(tname, nid string) (*events.Channel, func()) {
 
 	if tname != "" || nid != "" {
 		matcher = events.MatcherFunc(func(ev events.Event) bool {
-			var evt event
-			switch ev := ev.(type) {
-			case CreateEvent:
-				evt = event(ev)
-			case UpdateEvent:
-				evt = event(ev)
-			case DeleteEvent:
-				evt = event(ev)
-			}
-
-			if tname != "" && evt.Table != tname {
+			evt, ok := ev.(Event)
+			if !ok {
+				// FIXME(thaJeztah): verify this: the old code would fall-through, and return "true" ("Match")
 				return false
 			}
 
-			if nid != "" && evt.NetworkID != nid {
+			switch evt.Type {
+			case driverapi.Create, driverapi.Delete, driverapi.Update:
+				// ok
+				if tname != "" && evt.Table != tname {
+					return false
+				}
+				if nid != "" && evt.NetworkID != nid {
+					return false
+				}
+				return true
+			default:
+				// FIXME(thaJeztah): verify this: the old code would fall-through, and return "true" ("Match")
 				return false
```

Comment:
And here; although this code (before) was confusing, as it seemed to be returning random true/false for "other" type of events (i.e., if we didn't match it to any known type)

Reply:
Could be worse.
```go
matcher = events.MatcherFunc(func(ev events.Event) bool {
    evt, _ := ev.(Event)
    switch evt.Type {
    case driverapi.Create, driverapi.Delete, driverapi.Update:
        return true
    }
    return tname == "" || nid == ""
})
```